crowbar_framework/lib/crowbar/lock.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Lock
+    attr_accessor :logger
+    attr_accessor :path
+    attr_accessor :remote
+
+    def initialize(options = {})
+      @logger = options.fetch :logger, Rails.logger
+      @path = options.fetch :path, "/var/lock"
+      @remote = options.fetch :remote, false
+    end
+
+    def create(name, nodes = [])
+      if @remote
+        remote(name, nodes, action: "create")
+      else
+        local(name, action: "create")
```

Comment:
Rejected for concrete reasons which I clearly explained, not because I enjoy disputing comments / changes like you seem to be implying.  Yes, different classes makes sense when there is no easy way to reuse code between them.  Max will be having the fun though, not me, since I'm off on FTO now :)